graylog2-server/src/main/java/org/graylog2/shared/journal/KafkaJournal.java

```diff
@@ -168,9 +171,13 @@ public KafkaJournal(@Named("message_journal_dir") File journalDirectory,
         this.scheduler = scheduler;
         this.throttleThresholdPercentage = intRange(throttleThresholdPercentage, 0, 100);
         this.serverStatus = serverStatus;
+        this.maxSegmentSize = segmentSize.toBytes();
+        // Max message size should not be bigger than max segment size.
+        this.maxMessageSize = Ints.saturatedCast(maxSegmentSize);
 
         this.writtenMessages = metricRegistry.meter(name(this.getClass(), "writtenMessages"));
         this.readMessages = metricRegistry.meter(name(this.getClass(), "readMessages"));
+        this.writeDiscardedMessages = metricRegistry.meter(name(this.getClass(), "writeDiscardedMessages"));
```

Comment:
It's the other way around. We first write to the journal, then read. So in-flight means they are written to disk. A message can be discarded when writing to the journal and could be discarded when reading from the journal.